### PR TITLE
feat(frontend): search spans from extension tracers when object name is specified

### PR DESCRIPTION
**Pull request by xuqingyun**
### Description

- when searched object has no events during this period, try to discover any possible spans from extension tracers.

### CODE REVIEW

File: pkg/frontend/reader/merge/merge.go [hunk 1]
```diff
@@ -237,14 +237,24 @@ func newObject[M any](key objKey, trace *tftree.SpanTree, metadata M) (*object[M
 	if err != nil {
 		return nil, fmt.Errorf("clone spans: %w", err)
 	}
+
+	var m []M
+	if !isNil(metadata) {
```
[the remark this answers is not in the document]
Comment (xuqingyun) [reply pkg/frontend/reader/merge/merge.go:242]: @SOF3 if trace.Metadata is nil，slice长度为1，identifiers序列化之后变成了字符串null，读缓存的时候没有办法再去判断这个null是个什么东西。 trace.Metadata 改为slice是可以，不过在读取缓存的时候依然有这个风险遇到未知类型。
不过any不就是interface{}吗，所以M不就一定是个interface{}，如果不是interface不应该会编译错误？

[the remark this answers is not in the document]
Comment (SOF3) [reply pkg/frontend/reader/merge/merge.go:242]: `M` is `struct{}` when called from `GetTrace()`. `any(struct{}{}) == nil` is false.

[the remark this answers is not in the document]
Comment (SOF3) [reply pkg/frontend/reader/merge/merge.go:242]: I guess it meets your expectations, but checking metadata as `nil` randomly would be very strange (e.g. if `M` happens to be a thin pointer in the future, `isNil` will be false even for nil pointers).

I would prefer fixing this issue at a higher level, e.g. just making the `trace.Metadata` field an optional field or a slice.

[the remark this answers is not in the document]
Comment (SOF3) [reply pkg/frontend/reader/merge/merge.go]: Yes, but when `M` is `*Foo`, `any((*Foo)(nil)) == nil` is also false, which could be confusing.

[the remark this answers is not in the document]
Comment (SOF3) [reply pkg/frontend/reader/merge/merge.go]: I mean, from a generic point of view, `isNil` doesn't serve any semantic purpose, so this logic is quite specific to one of the callers anyway.

**Review by SOF3 — APPROVED** [review]
Approving as a temporary fix. We probably need to refactor the tracecache in the future anyway to allow better flexibility with the overall UX.

**Pull request merged**